### PR TITLE
Feat(challenges): Adds the ability to test with async methods

### PR DESCRIPTION
**Pull request by BerkeleyTrue**

### CODE REVIEW

File: client/frame-runner.js [hunk 1]
```diff
@@ -37,32 +38,68 @@ document.addEventListener('DOMContentLoaded', function() {
...
-              console.log('callback test');
+
+            if (helpers.isPromise(__result)) {
+              // turn promise into an observable
+              __result = Rx.Observable.fromPromise(__result);
             }
+          } else {
+            // test is not a function
+            // fill result with for compatibility
+            __result = Rx.Obserable.of(null);
```
Comment (Em-Ant) [inline client/frame-runner.js]: there's a typo here `Obserable`

File: client/frame-runner.js [hunk 2]
```diff
@@ -37,32 +38,68 @@ document.addEventListener('DOMContentLoaded', function() {
...
+              // We call it and store the result
+              // This result may be a promise or an observable or undefined
+              __result = test();
+            } else {
+              // if function takes arguments
+              // we expect it to be of the form
+              // function(cb) { /* ... */ }
+              // and callback has the following signature
+              // function(err) { /* ... */ }
+              __result = Rx.Observable.fromNodeCallback(test);
```
Comment (Em-Ant) [inline client/frame-runner.js]: I'm a rx newbie, but here I think you need this : 
` __result = Rx.Observable.fromNodeCallback(test)();`
I wasn't able to make the callback approach work, then I used the debugger and I noticed that `fromNodeCallback` returns a function, and this function returns the `Observable`.
Adding the `()` function call the callback approach works correctly

**Review by Em-Ant — CHANGES_REQUESTED** [review]
Async testing example https://hyperdev.com/#!/project/marsh-glazer

File: client/frame-runner.js (hunk 2 above)
Comment (BerkeleyTrue) [inline client/frame-runner.js]: Good catch!

**Comment by BerkeleyTrue** [comment]
@Em-Ant I've made those fixes. If the async stuff looks good please test to make sure this does not break normal challenges. Then it is safe to merge.

**Comment by Em-Ant** [comment]
@BerkeleyTrue @QuincyLarson  is importing the package.json tests right now. Then I think we will be able to run some experiments...

**Comment by BerkeleyTrue** [comment]
@Em-Ant Can you release this so that it can be merged in

**Comment by Em-Ant** [comment]
It should be ready this evening (tomorrow morning california time).

**Review by Em-Ant — APPROVED** [review]

**Pull request merged**